index.js

```diff
@@ -52,7 +53,7 @@ function wordLengths(str) {
 // ansi escape codes do not count towards length.
 function wrapWord(rows, word, cols) {
 	var insideEscape = false;
-	var visible = rows[rows.length - 1].length;
+	var visible = stripAnsi(rows[rows.length - 1]).length;
```

Comment:
Not sure how this works when hard wrapping, but I had to apply `strip-ansi` here in order to determine the correct visible characters. Otherwise it counts the invisible characters as well.